react/states_and_effects/introduction_to_state.md

```diff
@@ -0,0 +1,101 @@
+### Introduction
+
+Any exciting application you build is likely to change over its lifetime. The change could be as simple as toggling a dropdown menu or as complex as fetching data from an API. React provides primitives to manipulate state of your apps, more specifically components, to make them dynamic. In this lesson, we will learn how to use state in React.  
+
+Read  [What is State? by Academind](https://academind.com/tutorials/what-is-state) before we get started.
+
+### Lesson Overview
+
+This section contains a general overview of topics that you will learn in this lesson.
+
+* What is state?
+* How to use state in React.   
+* What happens when state changes in React?
+
+### What is State in React? 
+
+We write components in React, A LOT of components, and many times we want them to undergo visual changes as a result of user or computer interactions. For this purpose, a component needs to "remember" things about itself. This is where state comes in. **State is a component's memory.** 
+
+Let's take a look on how we can use state in our code.
+
+```
+CODESANDBOX EXAMPLE GOES HERE
+```
+
+An app that changes its background color based on the clicked button. Pretty simple, right? Take your time reading the code as we'll jump right in.
+
+#### The useState hook
+
+<span id="useState-hook">The `useState` hook is a built-in hook (we'll talk about hooks later) in React that allows you to define state in a functional component. It takes an initial value as a parameter and returns an array with two elements:</span>
```

Comment:
with two __elements:__ `to` with two __elements that we can destucture to get:__